scripts/github_pr/github_pr.rb

```diff
@@ -0,0 +1,405 @@
+#!/usr/bin/ruby
+
+require 'octokit'
+require 'optparse'
+require 'yaml'
+require 'json'
+
+class GithubAPI
+  def get_client
+    client = Octokit::Client.new(:netrc => true)
+    client.auto_paginate = true
+    client.login
+    client
+  end
+
+  def client
+    @@client ||= get_client
```

Comment:
Why not just using class methods? If you want to cache the client then this will not work.

Reply:
> Why not just using class methods?
Why should I?

> If you want to cache the client then this will not work.

Its a class variable. That should suffice.
Doing a test with a modified version with this patch:
https://gist.github.com/jdsn/1417cd744330a8110475aca8afd6112e
prints the same number over and over again. So its the same client object that is being used.